### ISSUE TITLE
Redesign concept of using libraries and headers (IDFGH-12880)

### ISSUE THREAD

**Issue by powerbroker** [espressif/esp-idf#13842]
### Answers checklist.

- [X] I have read the documentation [ESP-IDF Programming Guide](https://docs.espressif.com/projects/esp-idf/en/latest/) and the issue is not addressed there.
- [X] I have updated my IDF branch (master or release) to the latest version and checked that the issue is present there.
- [X] I have searched the issue tracker for a similar issue and not found a similar issue.

### General issue report

There are tons of 'generated' sources and libraries used implicitly: in Eclipse you have to enable eager indexing(which takes time) and rebuild index manually each time project is built.
Moreover there is no (simple) way to "attach" another sources instead of 'generated', where out-of-the-box-bugs are fixed and out-of-the-box-missing-features are implemented - to develop "industrial" way instead of "personal home toys" one.

ALL the used sources and libraries muzz be linked to project EXPLICITLY during project setup: so one person maintains driver project, another person - measurement filtering and calculations subsystem, third pirson - the application project, depending on that driver and measurements. Of course, there can be MANY applications, REUSING both driver and calculations being developed separately. C/C++ indexer within IDEs of that people doesn't overbloat because of eager indexing implicit dependencies overhead.
In other words, this way team development on ESP32 is not only theoretically possible, but flows transparently and easy - the same way, as on platforms with 10+ years experience background: STM32, AVR and even PIC. So, the great industrial potential of ESP32 hardware this way can be realized.

**Comment by igrr** [comment]
Hi @powerbroker,
In the [upcoming release](https://github.com/espressif/idf-eclipse-plugin/releases/tag/v3.0.0-beta) of idf-eclipse-plugin we will stop using "generated" sources as we are switching to a different indexer implementation based on clangd. With clangd, generated headers (except for the configuration header, sdkconfig.h) will no longer be necessary. Also, clangd updates the index much faster than the previous implementation.

I'll close this issue since it's not related to ESP-IDF itself but more to the IDE implementation. Please follow the idf-eclipse-plugin repository for future releases.
